File: Longest_Palindrome/Longest_Palindrome.py
def longest_palindrome(s):
    palindrome_list = []
    longest_palindrome = ""

    # Check all possible substrings
    for start in range(len(s)):
        for end in range(start + 1, len(s) + 1):
            substring = s[start:end]
            if substring == substring[::-1]:  # Check if the substring is a palindrome
                palindrome_list.append(substring)
                # Update longest_palindrome if the current one is longer
                if len(substring) > len(longest_palindrome):
                    longest_palindrome = substring

    print("Longest Palindromic Substring:", longest_palindrome)

# Test the function
longest_palindrome("babadd")
longest_palindrome("cbbd")
longest_palindrome("121")

File: Longest_Palindrome/Longest_Palindrome.java
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        longestPalindrome("babadd");
        longestPalindrome("cbbd");
        longestPalindrome("121");
    }

    public static void longestPalindrome(String s) {
        List<String> palindromeList = new ArrayList<>();
        String Longest_Palindrome = "";

        for (int start=0; start < s.length(); start++) {
            for (int end = start + 1 ; end <= s.length(); end++) {
                String substring = s.substring(start, end);
                if (substring.equals(new StringBuilder(substring).reverse().toString())) {
                    palindromeList.add(substring);
                    if (substring.length() > Longest_Palindrome.length()) {
                        Longest_Palindrome = substring;
                    }
                }
            }
        }

        System.out.println(Longest_Palindrome);
    }
}